packages/fxa-auth-server/lib/payments/iap/apple-app-store/apple-iap.ts

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Firestore } from '@google-cloud/firestore';
+import { AppStoreServerAPI, Environment } from 'app-store-server-api';
+import { Container } from 'typedi';
+import { TypedCollectionReference } from 'typesafe-node-firestore';
+
+import { AppConfig, AuthFirestore, AuthLogger } from '../../../types';
+import { IapConfig } from '../types';
+import { PurchaseManager } from './purchase-manager';
+
+export class AppleIAP {
+  private firestore: Firestore;
+  private iapConfigDbRef: TypedCollectionReference<IapConfig>;
+  private log: AuthLogger;
+  private prefix: string;
+
+  public purchaseManager: PurchaseManager;
+  constructor() {
+    const {
+      authFirestore,
+      subscriptions: { appStore },
+    } = Container.get(AppConfig);
+    this.prefix = `${authFirestore.prefix}iap-`;
+    this.firestore = Container.get(AuthFirestore);
+    this.iapConfigDbRef = this.firestore.collection(
+      `${this.prefix}iap-config`
+    ) as TypedCollectionReference<IapConfig>;
+    this.log = Container.get(AuthLogger);
+
+    // Initialize App Store Server API client per bundle ID
+    const appStoreServerApiClients = {};
+    for (const appName of Object.keys(appStore.credentials)) {
+      // Ideally, we'd key by bundleId instead of appName (e.g. 'org.mozilla.ios.FirefoxVPN'
+      // rather than 'guardian-vpn'), but we can't due to
+      // https://github.com/mozilla/node-convict/issues/250, so we will need to map appName
+      // to bundleId.
+      this.getBundleId(appName).then((bundleId) => {
```

Comment:
This was removed since I went with the underscore and string.replace approach.

If it were still in place, if this failed, the API client would not be initialized, and a check in the caller would have called the `initializeApiClientByBundleId` helper.